services/ingestion/engine.go

```diff
@@ -184,23 +184,21 @@ func (e *Engine) processTransactionEvent(event cadence.Event) error {
 		return fmt.Errorf("could not decode transaction event: %w", err)
 	}
 
-	// in case we have a direct call transaction we ignore it for now
-	// todo support indexing of direct calls
-	if tx == nil {
-		e.log.Debug().Str("event", event.String()).Msg("skipping direct call")
-		return nil
-	}
-
 	receipt, err := models.DecodeReceipt(event)
 	if err != nil {
 		return fmt.Errorf("failed to decode receipt: %w", err)
 	}
 
+	txHash, err := tx.Hash()
+	if err != nil {
+		return fmt.Errorf("failed to compute TX hash: %w", err)
+	}
+
```

Comment:
add a todo to remove this check once error is removed

Reply:
Added in https://github.com/onflow/flow-evm-gateway/pull/132/commits/5a48c66bd8ff4d562cb836256d0cb44044937737